include/swift/AST/Stmt.h

```diff
@@ -78,7 +78,6 @@ class alignas(8) Stmt {
   SourceLoc getEndLoc() const;
   
   SourceRange getSourceRange() const;
-  SourceLoc TrailingSemiLoc;
```

Comment:
I'm OK to preserve `TrailingSemiLoc`s. But I think we have a problem in the current implementation.

For example, on `let (a, b) = (1, 2)`, the parser produces 3 decls in this order:
```
      (pattern_binding_decl
        (pattern_tuple (pattern_named 'a') (pattern_named 'b'))
        (tuple_expr ...)
      (var_decl "a")
      (var_decl "b")
```
The current `master` implementation attaches `TrailingSemiLoc` only to the last `(var_decl "b")` which is not relevant, I think. It should be the `pattern_binding_decl`.

Fixing this is not trivial. If you are OK, I will do that in another PR.
Sounds good @bitjammer ?